File: Modul - 6/Soal - 5/PRAK605-2410817220009-NABILLAPUTRINUGRAHA.py
#fungsi buat mengkali dua matriks
def kali_matriks(A, B, n) :
    result = [[0] * n for _ in range(n)]
    for i in range(n) :
        for j in range(n) :
            for k in range(n) :
                result[i][j] += A[i][k] * B[k][j]
    return result

#buat ngnput ordo matriks
n = int(input(" "))

#buat nginput matriks A
print("Matriks A :")
A = [list(map(int, input().split()))for _ in range(n)]

#buat nginput matriks B
print("Matriks B :")
B = [list(map(int,input().split())) for _ in range(n)]

#buat ngitung hasil kali
result = kali_matriks(A, B, n)

#nyetak hasil
print("Matriks A x B :")
for baris in result :
    print(' '.join(map(str, baris)))

File: Modul - 6/Soal - 5/PRAK605-2410817220009-NABILLAPUTRINUGRAHA.c
#include <stdio.h>
int main() {
    int n;

    //buat nginput ordo matriks
    scanf("%d", &n);

    int A[n][n], B[n][n], result[n][n];

    //buat nginput matriks A
    printf("Matriks A : \n");
    for(int i = 0; i < n; i++) {
        for(int j = 0; j < n; j++) {
            scanf("%d", &A[i][j]);
        }
    }

    //buat nginput matriks B
    printf("Matriks B : \n");
    for(int i = 0; i < n; i++) {
        for(int j = 0; j < n; j++) {
            scanf("%d", &B[i][j]);
        }
    }

    //buat ngitung hasil kali matriks
    for(int i = 0; i < n; i++) {
        for(int j = 0; j < n; j++) {
            result[i][j] = 0;
            for(int k = 0; k < n; k++) {
                result[i][j] += A[i][k] * B[k][j];
            }
        }
    }

    //buat nyetak hasil
    printf("Matriks A x B : \n");
    for(int i = 0; i < n; i++) {
        for(int j = 0; j < n; j++) {
            printf("%d ", result[i][j]);
        }
        printf("\n");
    }
    return 0;
    
}